### ISSUE TITLE
Peculiar velocities or magnitudes

### ISSUE THREAD

**Issue by AlexGKim** [LSSTDESC/CCL#429]
An implementation of the peculiar velocity (or peculiar magnitude) correlation function.  Reference is Equation 29 in https://arxiv.org/pdf/1012.2912.pdf.

**Comment by AlexGKim** [comment]
Hi @philbull .  I suggest using as default the truncation given in Huterer et al. https://arxiv.org/pdf/1611.09862.pdf

We therefore assume `max = 20 if cos(θ) < 0.95 and `max = 200 otherwise 

Although the truncation could be user configurable.

I do not believe that any flat-sky approximation is used for the equation in question.

Finally, the application in question assumes spectroscopic redshifts, which effectively have delta-function uncertainties.  The science is not feasible in the LSST era with photo-z uncertainties.

**Comment by c-d-leonard** [comment]
@AlexGKim - can I ask if this is still something you would want? Sorry this issue has been hanging so long.

**Comment by damonge** [comment]
@c-d-leonard yes, this is something we still want to include, and the peculiar velocity folks have expressed more interest on it lately. I'd actually try to make this happen soon-ish

**Comment by c-d-leonard** [comment]
Okay, good to know. Do you think it would make sense to include it in the v3 milestone? (v3 is basically the proxy for the version we want for paper 2).

**Comment by damonge** [comment]
If we could do it quickly enough...

**Comment by AlexGKim** [comment]
Having it sooner than later would be great.  I am unfamiliar with CCL but if you need help let me know.

**Comment by damonge** [comment]
@AlexGKim Based on previous experience our main bottleneck is gonna be getting a benchmark (i.e. the observable calculated with a different code to compare with)

**Comment by AlexGKim** [comment]
@damonge .  For example, you would like the correlation function as a function of redshift for a fiducial cosmology?  I ask because I probably don't have the benchmark you are looking for (the reason I am interested in the CCL!) so we will have to engage others.

**Comment by c-d-leonard** [comment]
@AlexGKim yes that's about right - from an independent code. This can be anything from another existing public code base to something somebody has sitting on their laptop from a project five years ago and is willing to put public on github. I'm guessing someone in the supernova group will have at least some version of this coded up somewhere?

Also, just to check - I'm guessing this really does need the full bessel function treatment as in eq 29 of the linked paper since we are talking about the correlation for individual SN redshifts right? there's no equivalent to the Limber approximated version here?
